### PR TITLE
Add tests for RA-candidate bug

### PR DESCRIPTION
**Pull request by pablothedude**
Add test for RA candidate bug and fix some earlier on broken tests.

### CODE REVIEW

File: tests/behat/features/ra_bug-pieter.feature [hunk 1]
```diff
@@ -0,0 +1,33 @@
+Feature: A RA manages tokens tokens registered in the selfservice portal
+  In order to manage tokens
+  As a RA
+  I must be able to manage second factor tokens in RA
```
Comment (MKodde) [inline tests/behat/features/ra_bug-pieter.feature]: Please rewrite the feature description to match the issue that is verified in this feature. This seems to be a copy paste from ra_vet.feature

File: tests/behat/features/ra_bug-pieter.feature [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+  Scenario: Provision an institution and a user to promote later on by an authorized institution
+    Given institution "institution-a.example.com" can "select_raa" from institution "institution-a.example.com"
+      And institution "institution-a.example.com" can "use_ra" from institution "institution-a.example.com"
+      And institution "institution-d.example.com" can "use_raa" from institution "institution-a.example.com"
```
Comment (MKodde) [inline tests/behat/features/ra_bug-pieter.feature]: Maybe move this entry to the place where inst-d is configured? That makes it easier to follow how individual institutions are configured.

File: tests/behat/features/ra_bug-pieter.feature [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+      And institution "institution-f.example.com" can "select_raa" from institution "institution-f.example.com"
+      And institution "institution-f.example.com" can "use_ra" from institution "institution-a.example.com"
+      And institution "institution-f.example.com" can "use_ra" from institution "institution-b.example.com"
```
Comment (MKodde) [inline tests/behat/features/ra_bug-pieter.feature]: Why mention institution-b if it is not used in the scenario? Idem for line 19

File: tests/behat/features/ra_bug-pieter.feature [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    And the user "urn:collab:person:institution-a.example.com:joe-a-raa" has the role "raa" for institution "institution-a.example.com"
+
+      And a user "jane-d-raa institution-d.nl" identified by "urn:collab:person:institution-d.example.com:jane-d-raa" from institution "institution-d.example.com"
+      And the user "urn:collab:person:institution-d.example.com:jane-d-raa" has a vetted "yubikey"
+
+
+  Scenario: RAA from other institution can view the tokens page
+    Given I am logged in into the ra portal as "joe-a-raa" with a "yubikey" token
+  When I visit the RA Management page
+    Then I should see the following candidates:
```
Comment (MKodde) [inline tests/behat/features/ra_bug-pieter.feature]: Maybe add a step definition that reads

`Then I should see no candidates`

File: tests/behat/features/ra_bug-pieter.feature [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+      And a user "joe-a-raa institution-a" identified by "urn:collab:person:institution-a.example.com:joe-a-raa" from institution "institution-a.example.com"
+    And the user "urn:collab:person:institution-a.example.com:joe-a-raa" has a vetted "yubikey"
+    And the user "urn:collab:person:institution-a.example.com:joe-a-raa" has the role "raa" for institution "institution-a.example.com"
+
+      And a user "jane-d-raa institution-d.nl" identified by "urn:collab:person:institution-d.example.com:jane-d-raa" from institution "institution-d.example.com"
+      And the user "urn:collab:person:institution-d.example.com:jane-d-raa" has a vetted "yubikey"
+
+
+  Scenario: RAA from other institution can view the tokens page
```
Comment (MKodde) [inline tests/behat/features/ra_bug-pieter.feature]: Copy paste?

File: tests/behat/features/ra_bug-pieter.feature (hunk 2 above)
Comment (pablothedude) [reply tests/behat/features/ra_bug-pieter.feature]: This doesn't seem the latest version that's reviewed?

File: tests/behat/features/ra_bug-pieter.feature (hunk 3 above)
Comment (pablothedude) [reply tests/behat/features/ra_bug-pieter.feature]: This was in order to initially reproduce the bug report as close as possible. Removed it for now.

File: tests/behat/features/ra_bug-pieter.feature (hunk 4 above)
Comment (pablothedude) [reply tests/behat/features/ra_bug-pieter.feature]: This was already removed.

File: tests/behat/features/ra_bug-pieter.feature (hunk 5 above)
Comment (pablothedude) [reply tests/behat/features/ra_bug-pieter.feature]: This was already removed.

File: tests/behat/features/ra_candidate2.feature [hunk 6]
```diff
@@ -0,0 +1,30 @@
+Feature: A RAA manages ra candidates in the ra environment
+  In order to promote candidates
+  As a RAA
+  I must be able to promote and demote identities were I'm allowed to through the authorization config
+
+  Scenario: Provision an institution and a user to promote later on by an authorized institution
+    # Institution A
```
Comment (MKodde) [inline tests/behat/features/ra_candidate2.feature]: It seems your comments mention a scenario, but I'm not able to read more about this scenario. Ideas:

1. Add that scenario (or a link to it) to this feature
2. Update the comments to match the institutions that are actually being configured

File: tests/behat/features/ra_candidate2.feature [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    Given institution "institution-a.example.com" can "select_raa" from institution "institution-a.example.com"
+      And institution "institution-a.example.com" can "use_ra" from institution "institution-a.example.com"
+      And institution "institution-a.example.com" can "use_raa" from institution "institution-a.example.com"
+
+    # Institution V
+      And institution "institution-d.example.com" can "select_raa" from institution "institution-a.example.com"
+      And institution "institution-d.example.com" can "select_raa" from institution "institution-b.example.com"
```
Comment (MKodde) [inline tests/behat/features/ra_candidate2.feature]: Institution b reference seems unrelated? Or would this clash with default configuration?

File: tests/behat/features/ra_select-raa_use_raa.feature [hunk 8]
```diff
@@ -9,22 +9,22 @@ Feature: A RAA can only manage R RA(A)'s on the promotion page
...
     And institution "institution-a.example.com" can "use_raa" from institution "institution-a.example.com"
+    And institution "institution-a.example.com" can "use_raa" from institution "institution-d.example.com"
     And institution "institution-a.example.com" can "select_raa" from institution "institution-a.example.com"
     And institution "institution-a.example.com" can "select_raa" from institution "institution-d.example.com"
 
-    And institution "institution-d.example.com" can "use_ra" from institution "institution-d.example.com"
+    And institution "institution-d.example.com" can "use_raa" from institution "institution-a.example.com"
     And institution "institution-d.example.com" can "use_raa" from institution "institution-d.example.com"
-    And institution "institution-d.example.com" can "select_raa" from institution "institution-d.example.com"
+#    And institution "institution-d.example.com" can "select_raa" from institution "institution-d.example.com"
```
Comment (MKodde) [inline tests/behat/features/ra_select-raa_use_raa.feature]: (C|S)hould be removed completely?

File: tests/behat/features/ra_candidate2.feature (hunk 6 above)
Comment (pablothedude) [reply tests/behat/features/ra_candidate2.feature]: Removed the comments, because this is the scenario.

File: tests/behat/features/ra_candidate2.feature (hunk 7 above)
Comment (pablothedude) [reply tests/behat/features/ra_candidate2.feature]: With the current knowledge it indeed seems unrelated but initially it was impossible to conclude that.

File: tests/behat/features/ra_select-raa_use_raa.feature (hunk 8 above)
Comment (pablothedude) [reply tests/behat/features/ra_select-raa_use_raa.feature]: I deliberately left it there so the test would be better readable in the end.

File: tests/behat/features/ra_candidate2.feature (hunk 7 above)
Comment (MKodde) [reply tests/behat/features/ra_candidate2.feature]: Allright, makes sense!

File: tests/behat/features/ra_select-raa_use_raa.feature (hunk 8 above)
Comment (MKodde) [reply tests/behat/features/ra_select-raa_use_raa.feature]: Hmm, in that case an inverted step would be more explaining then a commented out step. Or give it some additional context by adding a comment. To me this does not add any readability.

Comment (pablothedude) [reply tests/behat/features/ra_select-raa_use_raa.feature]: Done
